MachineLearningPrototype/dataCollection/collect_data.py

```diff
@@ -0,0 +1,169 @@
+import requests
+import argparse
+from datetime import datetime
+from datetime import date
+from math import ceil
+import time
+from pprint import pprint
+import pymongo
+
+def get_num_pages(repo):
+	"""
+	Fetch the number of pages that store issues
+	"""
+	repo_query_url = f'https://api.github.com/repos/{repo}'
+	num_issues = requests.get(repo_query_url).json()['open_issues_count']
+	num_pages = ceil(num_issues/100)
+	return num_pages
+
+def store_in_db(data, collection_name, db, primary_key='url'):
+	"""
+	Store data in db.collection_name (update if exists using primary key)
+	"""
+	db_col = db[collection_name]
+	record_id = {primary_key: data[primary_key]}
+	db_col.update_one(record_id, {"$set": data}, upsert=True)
+
+def store_on_fs(data, file_name):
+	"""
+	Store data in file named `file_name`
+	"""
+	with open(file_name, "w") as f:
+		f.write(data)
+
+def get_collection_records(collection_name, db):
+	"""
+	Get all records of db.collection_name
+	"""
+	db_col = db[collection_name]
+	return db_col.find()
+
+def get_collection_record(query_params, field_params, collection_name, db):
+	"""
+	Get a record of db.collection_name specified by query_params and field_params
+	"""
+	db_col = db[collection_name]
+	return db_col.find_one(query_params, field_params)
+	
+def store_issue_details(issue, repo, db):
+	"""
+	Get issue details to be stored in DB and store issue text on filesystem
+	"""
+	issue_number = issue['number']
+	repo_name = repo.split('/')[-1]
+	issue_uid = f'{repo_name}_{issue_number}.txt'
+	issue_content_path = f'./data/GitHubData/IssueContent/{issue_uid}'
+	# test_output_path = f'./data/GitHubData/TestOutput/{issue_uid}'
+
+	#Get issue details
+	issue_details = dict()
+	issue_details['url'] = issue['html_url']
+	issue_details['repository_url'] = issue['repository_url']
+	issue_details['number'] = issue_number
+	issue_details['state'] = issue['state']
+	issue_details['title'] = issue['title']
+	issue_details['created_at'] = issue['created_at']
+	issue_details['updated_at'] = issue['updated_at']
+	issue_details['issue_content_path'] = issue_content_path
+	# issue_details['test_output_path'] = test_output_path
+
+	#Store issue content at issue_content_path
+	store_on_fs(issue['body'], issue_content_path)
+	
+	#Store issue details in db.Issues Table
+	store_in_db(issue_details, "Issues", db, "url")
+
+	#To do: Add issue_details['test_output_path'] and store test_output in filesystem
+
+
+def fetch_new_issues(repo, since, auth_token, db):
+	"""
+	Fetch all new open issues in the repo since `since`
+	"""
+	num_pages = get_num_pages(repo)
+	pprint(f'Need to query {num_pages} pages')
+
+	page_number = 1
+	num_open_issues = 0
```

Comment:
The `num_open_issues` here should be previous `total_issues_collected` in DB, otherwise saving this value into `total_issues_collected` in DB later will just store the most recently collected issue number, not the total number with previously collected issues.

Reply:
I have now changed the script to store the total number of issues collected (instead of the total number of new issues collected). Please let me know if that helps. Thank you!